src/test/java/com/exasol/github/AbstractGitHubGitRepositoryContentTest.java

```diff
@@ -0,0 +1,81 @@
+package com.exasol.github;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.*;
+
+import java.io.IOException;
+
+import org.junit.jupiter.api.Test;
+import org.kohsuke.github.*;
+import org.mockito.Mockito;
+
+import com.exasol.GitRepositoryContent;
+
+class AbstractGitHubGitRepositoryContentTest {
+    @Test
+    void testCreateAbstractGitHubGitRepositoryContentWithInvalidBranch() throws IOException {
+        final GHRepository ghRepositoryMock = Mockito.mock(GHRepository.class);
```

Comment:
Mocking 3rd-party classes is dangerous. That being said, I get why you are doing it in this case and it is acceptable.